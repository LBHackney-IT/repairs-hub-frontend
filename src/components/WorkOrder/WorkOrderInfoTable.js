import PropTypes from 'prop-types'
import { Table, TBody, TR, TH, TD } from '../Layout/Table'

const WorkOrderInfoTable = ({ workOrder }) => (
  <div>
    <Table className="govuk-!-margin-top-5 govuk-!-width-full">
      <TBody>
        <TR id="property">
          <TH scope="row">Property</TH>
          <TD>{workOrder.property}</TD>
        </TR>
        <TR id="tradeDescription">
          <TH scope="row">Trade</TH>
          <TD>{workOrder.tradeDescription}</TD>
        </TR>
        <TR id="description">
          <TH scope="row">Description</TH>
          <TD>{workOrder.description}</TD>
        </TR>
      </TBody>
    </Table>
  </div>
)

WorkOrderInfoTable.propTypes = {
  workOrder: PropTypes.object.isRequired,
}

export default WorkOrderInfoTable
