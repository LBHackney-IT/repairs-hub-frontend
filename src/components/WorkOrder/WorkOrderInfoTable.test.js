import { render } from '@testing-library/react'
import WorkOrderInfoTable from './WorkOrderInfoTable'

describe('WorkOrderInfoTable component', () => {
  const props = {
    workOrder: {
      reference: 10000012,
      tradeDescription: 'DOOR ENTRY ENGINEER - DE',
      description: 'Broken pipe in bathroom',
      property: '315 Banister House Homerton High Street',
    },
  }

  it('should render properly', () => {
    const { asFragment } = render(
      <WorkOrderInfoTable workOrder={props.workOrder} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
