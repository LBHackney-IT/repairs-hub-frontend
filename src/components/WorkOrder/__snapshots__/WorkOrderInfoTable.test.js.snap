// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkOrderInfoTable component should render properly 1`] = `
<DocumentFragment>
  <div>
    <table
      class="govuk-table lbh-table govuk-!-margin-top-5 govuk-!-width-full"
    >
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
          id="property"
        >
          <th
            class="govuk-table__header"
            scope="row"
          >
            Property
          </th>
          <td
            class="govuk-table__cell"
          >
            315 Banister House Homerton High Street
          </td>
        </tr>
        <tr
          class="govuk-table__row"
          id="tradeDescription"
        >
          <th
            class="govuk-table__header"
            scope="row"
          >
            Trade
          </th>
          <td
            class="govuk-table__cell"
          >
            DOOR ENTRY ENGINEER - DE
          </td>
        </tr>
        <tr
          class="govuk-table__row"
          id="description"
        >
          <th
            class="govuk-table__header"
            scope="row"
          >
            Description
          </th>
          <td
            class="govuk-table__cell"
          >
            Broken pipe in bathroom
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
